implementations/ruby-json_schemer/bowtie_json_schemer.rb

```diff
@@ -10,18 +10,34 @@ class UnsupportedCommand < StandardError; end
 class UnsupportedVersion < StandardError; end
 class UnsupportedDialect < StandardError; end
 
+JSON_SCHEMER_VERSION = Gem::Version.new(Gem::Specification.find_by_name('json_schemer').version)
+
 @meta_schema = nil
+@draft = nil
+
+if Gem::Version.new('2.0.0') <= JSON_SCHEMER_VERSION
+  @meta_schema_refs = {
+    JSONSchemer::Draft202012::BASE_URI => JSONSchemer::Draft202012::SCHEMA,
+    JSONSchemer::Draft201909::BASE_URI => JSONSchemer::Draft201909::SCHEMA,
+    JSONSchemer::Draft7::BASE_URI => JSONSchemer::Draft7::SCHEMA,
+    JSONSchemer::Draft6::BASE_URI => JSONSchemer::Draft6::SCHEMA,
+    JSONSchemer::Draft4::BASE_URI => JSONSchemer::Draft4::SCHEMA,
+  }
+  @meta_schema_refs.merge!(JSONSchemer::Draft202012::Meta::SCHEMAS)
+  @meta_schema_refs.merge!(JSONSchemer::Draft201909::Meta::SCHEMAS)
+  @meta_schema_refs.transform_keys! { |uri| uri.dup.tap { _1.fragment = nil } }
+else
+  @meta_schema_refs = {}
+end
 
-@meta_schema_refs = {
-  JSONSchemer::Draft202012::BASE_URI => JSONSchemer::Draft202012::SCHEMA,
-  JSONSchemer::Draft201909::BASE_URI => JSONSchemer::Draft201909::SCHEMA,
-  JSONSchemer::Draft7::BASE_URI => JSONSchemer::Draft7::SCHEMA,
-  JSONSchemer::Draft6::BASE_URI => JSONSchemer::Draft6::SCHEMA,
-  JSONSchemer::Draft4::BASE_URI => JSONSchemer::Draft4::SCHEMA,
-}
-@meta_schema_refs.merge!(JSONSchemer::Draft202012::Meta::SCHEMAS)
-@meta_schema_refs.merge!(JSONSchemer::Draft201909::Meta::SCHEMAS)
-@meta_schema_refs.transform_keys! { |uri| uri.dup.tap { _1.fragment = nil } }
+additional_dialects = if Gem::Version.new('2.0.0') <= JSON_SCHEMER_VERSION
```

Comment:
I'd try to do all this version stuff in one spot above.

Here you should be able to add `SUPPORTED_DIALECTS = ... above, and in an `else` (for post-2.0) do `SUPPORTED_DIALECTS = ...newerones`